### PR TITLE
Refactor the factors code handling

### PR DESCRIPTION
**Pull request by aarmey**
This makes a new class, `tensorGMM`, that handles a lot of the data wrangling. The goal of this is to consolidate a lot of the code so that it's harder to make mistakes and easier to explore results. Please look through this carefully before approving to merge.

### CODE REVIEW

File: gmm/tensor.py [hunk 1]
```diff
@@ -1,50 +1,159 @@
...
+        """Return covariance matrices."""
+        cov_chol = jnp.einsum(
+            "ax,bcx,dx,ex,fx->abcdef",
+            self.factors[0],
+            self.precisions,
+            *self.factors[2::],
+        )
+        return cov_chol
+
+    def sample(self, n_samples: int = 1000):
```
Comment (aarmey) [inline gmm/tensor.py:112]: Eventually (on a new PR) it would be nice to have this return an `xarray.Dataset`. This way all the axes labels would come along for the ride, and there would be far less repeated indexing to find the results of interest. Cyrillus recently spent some time learning to setup coupled datasets this way, so he might be able to advise.

File: gmm/tensor.py [hunk 2]
```diff
@@ -1,50 +1,159 @@
...
+        coords = {
+            "Cluster": np.arange(1, cp_factors.shape[0] + 1),
+            X.dims[0]: X.coords[X.dims[0]],
+            X.dims[2]: X.coords[X.dims[2]],
+            X.dims[3]: X.coords[X.dims[3]],
+            X.dims[4]: X.coords[X.dims[4]],
+        }
+
+        fac_df = [
+            pd.DataFrame(cp_factors.factors[ii], columns=cmpCol, index=coords[key])
```
Comment (aarmey) [inline gmm/tensor.py:148]: Eventually the factors should also be an `array.Dataset` rather than a dataframe, too. Almost all of the plotting code in all the figure files would only be a line or two, because xarray takes care of tracking all the labels.

**Comment by borcuttjahns** [comment]
Is this finished? If so I'll take a look through and play with it tonight.

**Comment by aarmey** [comment]
@borcuttjahns should be.

**Review by borcuttjahns — APPROVED** [review]
Played around with it and everything looks great and works much faster than before. Only thing I changed was fixing the centers of IL-2 reconstructed data centers. Should be good to merge pending Andrew's review! Thanks for the cleanup.

**Comment by aarmey** [comment]
Thanks. Yes, @andrewram4287, take a look through this sometime. I'm going to merge now.

File: gmm/tensor.py (hunk 1 above)
Comment (andrewram4287) [reply gmm/tensor.py:113]: Ok will be on that!

File: gmm/tensor.py (hunk 2 above)
Comment (andrewram4287) [reply gmm/tensor.py:145]: Will be on this too!
